### PR TITLE
Crate fix probes

### PR DESCRIPTION
**Pull request by modularTaco**
fixed liveness and readiness probes.
Previously the probes were running forever -> never succeeded -> prevented the pod from starting correctly

Replaced the curl which only looks at the headers with kubernetes native httpGet action.
The other one cannot be replaced with the httpGet action as it inspects the body of the response which httpGet is not able to do.

Additionally increased the supported cratedb version

### CODE REVIEW

**Comment by chicco785** [comment]
#97 will fix the build mechanism, once merged, you should rebase if everything works fine, we will then merge ;)

**Comment by c0c0n3** [comment]
@modularTaco #97 just got merged, if you could please rebase on master, thanks alot!!

File: charts/crate/values.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ app: crate
 
 image:
   name: crate
-  tag: 4.0.12
+  tag: 4.4.1
```
Comment (c0c0n3) [inline charts/crate/values.yaml:9]: can you change this to `4.6.7`?

File: charts/crate/README.MD [hunk 2]
```diff
@@ -42,7 +42,7 @@ The following table lists the configurable parameters of the cratedb chart and t
 |-----------------------------------|---------------------------------------------|-----------------------------------------------------|
 | `replicaCount`                    | Amount of pods to spawn                     | `3`                                                 |
 | `image.name`                      | The image to pull                           | `crate`                                             |
-| `image.tag`                       | The version of the image to pull            | `4.2.1`                                             |
+| `image.tag`                       | The version of the image to pull            | `4.4.1`                                             |
```
Comment (c0c0n3) [inline charts/crate/README.MD:45]: can you change the version to `4.6.7`?

**Comment by modularTaco** [comment]
Hi @c0c0n3 ,

i've rebased the branch and set the supported version fo cratedb to 4.6.7.  
Unfortunately the PR is more than a year old and a lot of things happen in one year. In my case i've left my previous employer and do not have access to any k8s infrastructure anymore to retest this PR with crate 4.6.7. I do not expect any problems and i'm pretty sure i ran this version of the helm chart locally with crate 4.6.7 in already february, but i cannot confirm securely.

**Comment by c0c0n3** [comment]
hi @modularTaco :-)

> i've rebased the branch and set the supported version fo cratedb to 4.6.7.

good stuff!

> left my previous employer and do not have access to any k8s infrastructure anymore to retest

no worries, we'll test this. thanks so much for still putting in the time to wrap up, even if you're not paid for it anymore. bless you!

**Comment by c0c0n3** [comment]
@modularTaco so I've managed to test the liveness and readiness probes, they both work flawlessly 👍 
I checked both success and failure scenarios

```console
$ kubectl describe pod/release-name-crate-0
Name:         release-name-crate-0
...
Status:       Running
...
Containers:
  crate:
    State:          Running
      Started:      Thu, 21 Jul 2022 12:04:17 +0200
    Last State:     Terminated
      Reason:       Error
      Exit Code:    78
      Started:      Thu, 21 Jul 2022 12:03:22 +0200
      Finished:     Thu, 21 Jul 2022 12:03:29 +0200
    Ready:          True
    Restart Count:  4
    Liveness:       http-get http://:crate-web/ delay=240s timeout=5s period=10s #success=1 #failure=5
    Readiness:      exec [sh -c curl -s http://localhost:4200 | grep -q '"ok" : true,'] delay=60s timeout=1s period=60s #success=1 #failure=3
...
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
...
Events:
  Type     Reason     Age                    From                Message
  ----     ------     ----                   ----                -------
  Normal   Scheduled  4m53s                  default-scheduler   Successfully assigned default/release-name-crate-0 to orchestra
  Normal   Pulling    4m53s                  kubelet, orchestra  Pulling image "crate:4.6.7"
  Normal   Pulled     4m24s                  kubelet, orchestra  Successfully pulled image "crate:4.6.7"
  Warning  BackOff    2m31s (x9 over 4m4s)   kubelet, orchestra  Back-off restarting failed container
  Normal   Created    2m18s (x5 over 4m24s)  kubelet, orchestra  Created container crate
  Normal   Started    2m18s (x5 over 4m23s)  kubelet, orchestra  Started container crate
  Normal   Pulled     2m18s (x4 over 4m12s)  kubelet, orchestra  Container image "crate:4.6.7" already present on machine
```

Helm: 2.17.0
K8s: 1.16.15

**Review by c0c0n3 — APPROVED** [review]

**Comment by c0c0n3** [comment]
@modularTaco thanks a stack for contributing this PR, much appreciated!!

**Comment by modularTaco** [comment]
Perfect, thank you! 👌

**Pull request merged**